auth-api/tests/unit/services/test_flags.py

```diff
@@ -0,0 +1,93 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test-Suite to ensure that the Flag Service is working as expected."""
+import pytest
+from flask import Flask
+
+from auth_api.services import Flags
+
+app = None
+
+
+@pytest.fixture
+def setup():
+    """Initialize app with dev env for testing."""
+    global app
+    app = Flask(__name__)
+    app.env = 'development'
+
+
+def test_flags_constructor_no_app(setup):
+    """Ensure that flag object can be initialized."""
+    flags = Flags()
+    assert flags
+
+
+def test_flags_constructor_with_app(setup):
+    """Ensure that extensions can be initialized."""
+    with app.app_context():
+        flags = Flags(app)
+    assert flags
+    assert app.extensions['featureflags']
+
+
+def test_init_app_dev_with_key(setup):
+    """Ensure that extension can be initialized with a key in dev."""
+    app.config['AUTH_LD_SDK_KEY'] = 'https://no.flag/avail'
+
+    with app.app_context():
+        flags = Flags()
+        flags.init_app(app)
+    assert flags
+    assert app.extensions['featureflags']
+    assert app.extensions['featureflags'].get_sdk_key() == 'https://no.flag/avail'
+
+
+def test_init_app_dev_no_key(setup):
+    """Ensure that extension can be initialized with no key in dev."""
+    app.config['AUTH_LD_SDK_KEY'] = None
+
+    with app.app_context():
+        flags = Flags()
+        flags.init_app(app)
+    assert flags
+    assert app.extensions['featureflags']
+
+
+def test_init_app_prod_with_key(setup):
+    """Ensure that extension can be initialized with a key in prod."""
+    app.env = 'production'
+    app.config['AUTH_LD_SDK_KEY'] = 'https://no.flag/avail'
+
+    with app.app_context():
+        flags = Flags()
+        flags.init_app(app)
+    assert flags
+    assert app.extensions['featureflags']
+    assert app.extensions['featureflags'].get_sdk_key() == 'https://no.flag/avail'
+
+
+def test_init_app_prod_no_key(setup):
+    """Ensure that extension can be initialized with no key in prod."""
+    app.env = 'production'
+    app.config['AUTH_LD_SDK_KEY'] = None
+
+    with app.app_context():
+        flags = Flags()
+        flags.init_app(app)
+        with pytest.raises(KeyError):
+            client = app.extensions['featureflags']
+            assert not client
+    assert flags
```

Comment:
Could you add in some tests here to assert the flag values?

In https://github.com/bcgov/registries-search we also have some integration test examples (you need the LD flag set for them) which would be great to have. They can be pretty useful when experimenting with new flags especially when adding segments for specific users, etc.

Reply:
Was just looking at them and they don't test much haha, but they're a good stepping stone for flag tests once real flags are added to the api and targeting specific users

Reply:
yup added